maven-repository-manager/src/main/java/org/jboss/pnc/mavenrepositorymanager/RepositoryManagerDriver.java

```diff
@@ -262,8 +266,47 @@ private void setupBuildRepos(BuildExecution execution, Indy indy)
             // Global-level repos, for captured/shared artifacts and access to the outside world
             addGlobalConstituents(buildGroup);
 
-            indy.stores().create(buildGroup,
-                    "Creating repository group for resolving artifacts in build: " + id + " (repo: " + buildContentId + ")", Group.class);
+            // add the repositories removed from poms by the adjust process
+            List<ArtifactRepository> repositories = execution.getArtifactRepositories();
+            if (repositories != null && !repositories.isEmpty()) {
+                StoreListingDTO<RemoteRepository> existingRepos = indy.stores().listRemoteRepositories(MAVEN_PKG_KEY);
+                for (ArtifactRepository repository : repositories) {
+                    StoreKey remoteKey = null;
+                    for (RemoteRepository existingRepo : existingRepos) {
+                        if (existingRepo.getUrl().equals(repository.getUrl())) {
+                            remoteKey = existingRepo.getKey();
+                            break;
+                        }
+                    }
+
+                    if (remoteKey == null) {
+                        // this is basically an implied repo, so using the same prefix "i-"
+                        String remoteName = "i-" + repository.getId();
+
+                        // find a free repository ID for the newly created repo
+                        remoteKey = new StoreKey(MAVEN_PKG_KEY, StoreType.remote, remoteName);
+                        int i = 2;
+                        while (indy.stores().exists(remoteKey)) {
+                            remoteKey = new StoreKey(MAVEN_PKG_KEY, StoreType.remote, remoteName + "-" + i++);
+                        }
+
+                        RemoteRepository remoteRepo = new RemoteRepository(MAVEN_PKG_KEY, remoteKey.getName(), repository.getUrl());
+                        remoteRepo.setAllowReleases(repository.getReleases());
+                        remoteRepo.setAllowSnapshots(repository.getSnapshots());
+                        remoteRepo.setDescription("Implicitly created repo for: " + repository.getName() + " ("
+                                + repository.getId() + ") from repository declaration removed by PME in build "
+                                + id + " (repo: " + buildContentId + ")");
+                        indy.stores().create(remoteRepo,
+                                "Creating hosted repository for build: " + id + " (repo: " + buildContentId + ")",
+                                RemoteRepository.class);
+                    }
+
+                    buildGroup.addConstituent(remoteKey);
+                }
+            }
+
+            indy.stores().create(buildGroup, "Creating repository group for resolving artifacts in build: " + id
+                    + " (repo: " + buildContentId + ")", Group.class);
```

Comment:
To improve readability it would be nice to extract some methods out of this long one.